db-engine/src/main/java/com/splicemachine/db/iapi/types/UserType.java

```diff
@@ -723,22 +708,11 @@ public void read(Row row, int ordinal) throws StandardException {
 			setToNull();
 		else {
 			isNull = false;
-			byte[] bytes = (byte[]) row.get(ordinal);
-			ByteArrayInputStream bis = new ByteArrayInputStream(bytes);
-			ObjectInput in = null;
-			try {
-				in = new ObjectInputStream(bis);
-				value = in.readObject();
-			} catch (Exception e) {
-				throw new RuntimeException(e);
-			} finally {
-				try {
-					if (in != null) {
-						in.close();
-					}
-				} catch (IOException ex) {
-					// ignore close exception
-				}
+			Object object = row.get(ordinal);
+			if (object instanceof byte[]) {
+				value = SerializationUtils.deserialize((byte[]) row.get(ordinal));
```

Comment:
Don't you want to pass `object` to `deserialize` instead of calling `row.get(ordinal)` a second time?